activation/activation_test.go

```diff
@@ -116,25 +116,31 @@ func (ms *MockSigning) Sign(m []byte) []byte {
 
 type NIPostBuilderMock struct {
 	poetRef         []byte
-	buildNIPostFunc func(challenge *types.Hash32) (*types.NIPost, error)
+	buildNIPostFunc func(challenge *types.Hash32) (*types.NIPost, time.Duration, error)
 	SleepTime       int
 }
 
-func (np NIPostBuilderMock) updatePoETProver(PoetProvingServiceClient) {}
+// A compile time check to ensure that `NIPostBuilderMock` fully implements the `nipostBuilder` interface.
+var _ nipostBuilder = (*NIPostBuilderMock)(nil)
 
-func (np *NIPostBuilderMock) BuildNIPost(_ context.Context, challenge *types.Hash32, _ chan struct{}) (*types.NIPost, error) {
+func (np NIPostBuilderMock) updatePoETProvers([]PoetProvingServiceClient) {}
+
+func (np *NIPostBuilderMock) BuildNIPost(_ context.Context, challenge *types.Hash32, _ chan struct{}) (*types.NIPost, time.Duration, error) {
 	if np.buildNIPostFunc != nil {
 		return np.buildNIPostFunc(challenge)
 	}
-	return newNIPostWithChallenge(challenge, np.poetRef), nil
+	return newNIPostWithChallenge(challenge, np.poetRef), 0, nil
 }
 
 type NIPostErrBuilderMock struct{}
 
-func (np *NIPostErrBuilderMock) updatePoETProver(PoetProvingServiceClient) {}
+// A compile time check to ensure that `NIPostErrBuilderMock` fully implements the `nipostBuilder` interface.
+var _ nipostBuilder = (*NIPostErrBuilderMock)(nil)
```

Comment:
This check is not needed, if the mock wouldn't implement the interface the compiler already tells you so; no need for an extra check here.